python/seldon_core/metadata.py

```diff
@@ -137,6 +137,7 @@ class SeldonInvalidMetadataError(Exception):
         "platform": {"type": "string"},
         "inputs": INPUTS_OUTPUTS_SCHEMA,
         "outputs": INPUTS_OUTPUTS_SCHEMA,
+        "custom": {"type": "object", "additionalProperties": {"type": "string"}},
```

Comment:
Ok so the only restriction is that it would be restricted to only key value strings? So basically if it was ints or lists it would fail?

Reply:
Yes, behaviour would be consistent with other fields not respecting the schema.
Basically MICROSERVICE_BAD_METADATA would be return on the  endpoint as described in the [docs](https://docs.seldon.io/projects/seldon-core/en/latest/examples/metadata.html#3.-Invalid-metadata-definition). 

Also in the logs there would be quite verbose information why the definition does not fulfil the schema.